File: src/main/java/org/quang1807/citizenstrade/managers/EconomyManager.java
package org.quang1807.citizenstrade.managers;

import org.quang1807.citizenstrade.CitizenShop;
import net.milkbowl.vault.economy.Economy;
import org.black_ixx.playerpoints.PlayerPoints;
import org.black_ixx.playerpoints.PlayerPointsAPI;
import org.bukkit.entity.Player;
import org.bukkit.plugin.RegisteredServiceProvider;

public class EconomyManager {
    private final CitizenShop plugin;
    private Economy economy;
    private PlayerPointsAPI playerPointsAPI;

    public EconomyManager(CitizenShop plugin) {
        this.plugin = plugin;
    }

    public boolean setupEconomy() {
        if (plugin.getServer().getPluginManager().getPlugin("Vault") == null) {
            return false;
        }

        RegisteredServiceProvider<Economy> rsp = plugin.getServer().getServicesManager().getRegistration(Economy.class);
        if (rsp == null) {
            return false;
        }

        economy = rsp.getProvider();
        return economy != null;
    }

    public boolean setupPlayerPoints() {
        if (plugin.getServer().getPluginManager().getPlugin("PlayerPoints") == null) {
            return false;
        }

        try {
            playerPointsAPI = PlayerPoints.getInstance().getAPI();
            return playerPointsAPI != null;
        } catch (Exception e) {
            plugin.getLogger().warning("Error setting up PlayerPoints: " + e.getMessage());
            return false;
        }
    }

    // Economy methods
    public boolean hasEnoughMoney(Player player, double amount) {
        return economy != null && economy.has(player, amount);
    }

    public void takeMoney(Player player, double amount) {
        if (economy != null) {
            economy.withdrawPlayer(player, amount);
        }
    }

    public void giveMoney(Player player, double amount) {
        if (economy != null) {
            economy.depositPlayer(player, amount);
        }
    }

    public double getBalance(Player player) {
        return economy != null ? economy.getBalance(player) : 0.0;
    }

    // PlayerPoints methods
    public boolean hasEnoughPoints(Player player, int amount) {
        return playerPointsAPI != null && playerPointsAPI.look(player.getUniqueId()) >= amount;
    }

    public void takePoints(Player player, int amount) {
        if (playerPointsAPI != null) {
            playerPointsAPI.take(player.getUniqueId(), amount);
        }
    }

    public void givePoints(Player player, int amount) {
        if (playerPointsAPI != null) {
            playerPointsAPI.give(player.getUniqueId(), amount);
        }
    }

    public int getPoints(Player player) {
        return playerPointsAPI != null ? playerPointsAPI.look(player.getUniqueId()) : 0;
    }

    // Getters
    public Economy getEconomy() {
        return economy;
    }

    public PlayerPointsAPI getPlayerPointsAPI() {
        return playerPointsAPI;
    }

    public boolean isEconomyEnabled() {
        return economy != null;
    }

    public boolean isPlayerPointsEnabled() {
        return playerPointsAPI != null;
    }
}

File: src/main/kotlin/org/quang1807/fancyshop/managers/EconomyManager.kt
package org.quang1807.fancyshop.managers

import org.quang1807.fancyshop.FancyShop
import net.milkbowl.vault.economy.Economy
import org.black_ixx.playerpoints.PlayerPointsAPI
import org.bukkit.plugin.RegisteredServiceProvider

class EconomyManager(private val plugin: FancyShop) {

    var economy: Economy? = null
        private set
    var playerPointsAPI: PlayerPointsAPI? = null
        private set

    fun setupEconomy(): Boolean {
        if (plugin.server.pluginManager.getPlugin("Vault") == null) {
            return false
        }

        val rsp: RegisteredServiceProvider<Economy>? = plugin.server.servicesManager.getRegistration(Economy::class.java)
        if (rsp == null) {
            return false
        }

        economy = rsp.provider
        return economy != null
    }

    fun setupPlayerPoints() {
        val plugin = this.plugin.server.pluginManager.getPlugin("PlayerPoints")
        if (plugin != null && plugin is org.black_ixx.playerpoints.PlayerPoints) {
            try {
                playerPointsAPI = plugin.api
                this.plugin.logger.info("PlayerPoints integration enabled!")
            } catch (e: Exception) {
                this.plugin.logger.warning("Failed to hook into PlayerPoints: ${e.message}")
            }
        }
    }
}